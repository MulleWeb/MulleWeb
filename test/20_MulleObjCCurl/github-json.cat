#! /bin/sh

sed -e 's/[A-Za-z0-9_-]*\.github[a-z]*\.com/xxx.github.com/g' "$@"
