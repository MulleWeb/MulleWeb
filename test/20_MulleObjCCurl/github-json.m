#import <MulleWebClient/MulleWebClient.h>


static NSString  *URL = @"https://api.github.com/users/mulle-nat/orgs";


int   main( int argc, const char * argv[])
{
   MulleCurl         *curl;
   MulleJSMNParser   *parser;
   id                plist;
   NSError           *error;

#ifdef __MULLE_OBJC__
   if( mulle_objc_global_check_universe( __MULLE_OBJC_UNIVERSENAME__) != mulle_objc_universe_is_ok)
   {
      fprintf( stderr, "Stuck classes or categories\n");
      return( 22);
   }
#endif

   curl   = [MulleCurl object];
   parser = [MulleJSMNParser object];

   [curl setParser:parser];
   [curl setRequestHeaders:@{
                              @"Accept" : @"application/vnd.github.v3+json",
                              @"User-Agent": @"Awesome-Octocat-App"  // :=)
                            }];
   // for testing turn off https certi checks, which are not ez
   // crossplatform wise
   [curl setOptions:@{
                       @"CURLOPT_SSL_VERIFYPEER": @(NO),
                       @"CURLOPT_SSL_VERIFYHOST": @(NO)
                     }];

   plist = [curl parseContentsOfURLWithString:URL];
   if( ! plist)
   {
      error = [NSError mulleExtract];
      fprintf( stderr, "Error: %s\n", [[error description] UTF8String]);
      return( 1);
   }

   printf( "%s\n", [[plist description] UTF8String]);
   return( 0);
}
