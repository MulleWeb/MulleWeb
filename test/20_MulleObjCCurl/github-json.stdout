(
    {
        "avatar_url" = "https://xxx.github.com/u/6410439\?v=4";
        description = "";
        "events_url" = "https://xxx.github.com/orgs/mulle-kybernetik/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-kybernetik/hooks";
        id = 6410439;
        "issues_url" = "https://xxx.github.com/orgs/mulle-kybernetik/issues";
        login = "mulle-kybernetik";
        "members_url" = "https://xxx.github.com/orgs/mulle-kybernetik/members{/member}";
        "node_id" = "MDEyOk9yZ2FuaXphdGlvbjY0MTA0Mzk=";
        "public_members_url" = "https://xxx.github.com/orgs/mulle-kybernetik/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-kybernetik/repos";
        url = "https://xxx.github.com/orgs/mulle-kybernetik";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/23309093\?v=4";
        description = "An Objective-C runtime based on mulle-core and mulle-clang. #MakeObjCGreatAgain";
        "events_url" = "https://xxx.github.com/orgs/mulle-objc/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-objc/hooks";
        id = 23309093;
        "issues_url" = "https://xxx.github.com/orgs/mulle-objc/issues";
        login = "mulle-objc";
        "members_url" = "https://xxx.github.com/orgs/mulle-objc/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjIzMzA5MDkz;
        "public_members_url" = "https://xxx.github.com/orgs/mulle-objc/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-objc/repos";
        url = "https://xxx.github.com/orgs/mulle-objc";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/35078660\?v=4";
        description = "The Unix IDE. Written in bash.";
        "events_url" = "https://xxx.github.com/orgs/mulle-sde/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-sde/hooks";
        id = 35078660;
        "issues_url" = "https://xxx.github.com/orgs/mulle-sde/issues";
        login = "mulle-sde";
        "members_url" = "https://xxx.github.com/orgs/mulle-sde/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjM1MDc4NjYw;
        "public_members_url" = "https://xxx.github.com/orgs/mulle-sde/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-sde/repos";
        url = "https://xxx.github.com/orgs/mulle-sde";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/35923740\?v=4";
        description = "The Foundation library for mulle-objc";
        "events_url" = "https://xxx.github.com/orgs/MulleFoundation/events";
        "hooks_url" = "https://xxx.github.com/orgs/MulleFoundation/hooks";
        id = 35923740;
        "issues_url" = "https://xxx.github.com/orgs/MulleFoundation/issues";
        login = MulleFoundation;
        "members_url" = "https://xxx.github.com/orgs/MulleFoundation/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjM1OTIzNzQw;
        "public_members_url" = "https://xxx.github.com/orgs/MulleFoundation/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/MulleFoundation/repos";
        url = "https://xxx.github.com/orgs/MulleFoundation";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/35930577\?v=4";
        description = "Pure C libraries (no system calls, no threads, no atomics)";
        "events_url" = "https://xxx.github.com/orgs/mulle-c/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-c/hooks";
        id = 35930577;
        "issues_url" = "https://xxx.github.com/orgs/mulle-c/issues";
        login = "mulle-c";
        "members_url" = "https://xxx.github.com/orgs/mulle-c/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjM1OTMwNTc3;
        "public_members_url" = "https://xxx.github.com/orgs/mulle-c/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-c/repos";
        url = "https://xxx.github.com/orgs/mulle-c";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/43937304\?v=4";
        description = "C libraries for concurrent programming (threads, atomics) based on mulle-c.";
        "events_url" = "https://xxx.github.com/orgs/mulle-concurrent/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-concurrent/hooks";
        id = 43937304;
        "issues_url" = "https://xxx.github.com/orgs/mulle-concurrent/issues";
        login = "mulle-concurrent";
        "members_url" = "https://xxx.github.com/orgs/mulle-concurrent/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjQzOTM3MzA0;
        "public_members_url" = "https://xxx.github.com/orgs/mulle-concurrent/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-concurrent/repos";
        url = "https://xxx.github.com/orgs/mulle-concurrent";
    },
    {
        "avatar_url" = "https://xxx.github.com/u/44543973\?v=4";
        description = "C libraries based on mulle-c and mulle-concurrent";
        "events_url" = "https://xxx.github.com/orgs/mulle-core/events";
        "hooks_url" = "https://xxx.github.com/orgs/mulle-core/hooks";
        id = 44543973;
        "issues_url" = "https://xxx.github.com/orgs/mulle-core/issues";
        login = "mulle-core";
        "members_url" = "https://xxx.github.com/orgs/mulle-core/members{/member}";
        "node_id" = MDEyOk9yZ2FuaXphdGlvbjQ0NTQzOTcz;
        "public_members_url" = "https://xxx.github.com/orgs/mulle-core/public_members{/member}";
        "repos_url" = "https://xxx.github.com/orgs/mulle-core/repos";
        url = "https://xxx.github.com/orgs/mulle-core";
    }
)
